src/bindings/python/src/pyopenvino/frontend/node_context.cpp

```diff
@@ -0,0 +1,127 @@
+            if (!(py_type).is_none()) {       \
+                return py_type(casted);       \
+            }                                 \
+            return py::cast(casted);          \
+        }                                     \
+    }
+
+void regclass_frontend_NodeContext(py::module m) {
+    py::class_<ov::frontend::NodeContext, std::shared_ptr<ov::frontend::NodeContext>> ext(m,
+                                                                                          "NodeContext",
+                                                                                          py::dynamic_attr());
+
+    ext.def(
+        "get_attribute",
+        [=](NodeContext& self, const std::string& name, const py::object& default_value, const py::object& dtype)
+            -> py::object {
+            auto any = self.get_attribute_as_any(name);
+            auto module = py::module_::import("openvino.runtime");
+
+            auto type = m.attr("Type");
+            if (dtype == type) {
+                if (any.is<int32_t>() || any.is<int64_t>()) {
+                    return py::cast(self.get_attribute<ov::element::Type>(name));
+                } else if (any.is<std::vector<int32_t>>() || any.is<std::vector<int64_t>>()) {
+                    return py::cast(self.get_attribute<std::vector<ov::element::Type>>(name));
+                }
+            }
+
+            CAST_TO_PY(any, dtype, int32_t);
+            CAST_TO_PY(any, dtype, int64_t);
+            CAST_TO_PY(any, dtype, bool);
+            CAST_TO_PY(any, dtype, std::string);
+            CAST_TO_PY(any, dtype, float);
+            CAST_TO_PY(any, dtype, ov::element::Type);
+            CAST_TO_PY(any, dtype, ov::PartialShape);
+
+            CAST_VEC_TO_PY(any, dtype, std::vector<int32_t>);
+            CAST_VEC_TO_PY(any, dtype, std::vector<int64_t>);
+#ifndef __APPLE__
+            // TODO: investigate the issue in pybind11 on MacOS
+            CAST_VEC_TO_PY(any, dtype, std::vector<bool>);
+#endif
+            CAST_VEC_TO_PY(any, dtype, std::vector<std::string>);
+            CAST_VEC_TO_PY(any, dtype, std::vector<float>);
+            CAST_VEC_TO_PY(any, dtype, std::vector<ov::element::Type>);
+            CAST_VEC_TO_PY(any, dtype, std::vector<ov::PartialShape>);
+
+            if (default_value.is_none())
+                FRONT_END_GENERAL_CHECK(false, "Attribute ", name, " can't be converted to defined types.");
+            else
+                return default_value;
+        },
+        py::arg("name"),
+        py::arg("default_value") = py::none(),
+        py::arg("dtype") = py::none());
+
+    ext.def("get_input", [](NodeContext& self, int idx) {
+        return self.get_input(idx);
+    });
+
+    ext.def("get_input", [](NodeContext& self, const std::string& name) {
+        return self.get_input(name);
+    });
+
+    ext.def("get_input", [](NodeContext& self, const std::string& name, int idx) {
+        return self.get_input(name, idx);
+    });
```

Comment:
```suggestion
    ext.def("get_input", [](NodeContext& self, const std::string& name, int idx) {
        return self.get_input(name, idx);
    }, py::arg("name"), py::arg("idx"));
```
`py::arg` can be added to all functions here.

Reply:
I will cover documentation and these naming topics in the next PR